biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java

```diff
@@ -561,8 +582,7 @@ private TaggedData doConnect(Object put, Type ref, final URLConnection con, fina
 			// origin server presented.
 			return new TaggedData(request.url.toURI(), 526, request.useCacheFile);
 		} catch (SocketTimeoutException ste) {
-			task.done(ste.toString(), null);
-			return new TaggedData(request.url.toURI(), HTTP_GATEWAY_TIMEOUT, request.useCacheFile);
+			throw new TimeoutException();
```

Comment:
It may be useful to make `ste` be the cause of the TimeoutException. Also, since you convert the TimeoutException to a string in the catch clause above, a message for the exception may be useful here.

Reply:
There is nothing interesting the exception